test/core/security/security_connector_test.cc

```diff
@@ -161,66 +161,51 @@ static int check_identity(const grpc_auth_context* ctx,
   return 1;
 }
 
-static int check_x509_cn(const grpc_auth_context* ctx,
-                         const char* expected_cn) {
-  grpc_auth_property_iterator it = grpc_auth_context_find_properties_by_name(
-      ctx, GRPC_X509_CN_PROPERTY_NAME);
+static int check_property_with_single_value(
+    const grpc_auth_context* ctx, const char* expected_property_name,
+    const char* expected_property_value) {
+  grpc_auth_property_iterator it =
+      grpc_auth_context_find_properties_by_name(ctx, expected_property_name);
   const grpc_auth_property* prop = grpc_auth_property_iterator_next(&it);
   if (prop == nullptr) {
-    gpr_log(GPR_ERROR, "CN property not found.");
+    gpr_log(GPR_ERROR, "Expected value %s not found.", expected_property_value);
     return 0;
   }
-  if (strncmp(prop->value, expected_cn, prop->value_length) != 0) {
-    gpr_log(GPR_ERROR, "Expected CN %s and got %s", expected_cn, prop->value);
+  if (strncmp(prop->value, expected_property_value, prop->value_length) != 0) {
+    gpr_log(GPR_ERROR, "Expected value %s and got %s for property %s.",
+            expected_property_value, prop->value, expected_property_name);
     return 0;
   }
   if (grpc_auth_property_iterator_next(&it) != nullptr) {
-    gpr_log(GPR_ERROR, "Expected only one property for CN.");
+    gpr_log(GPR_ERROR, "Expected only one property for %s.",
+            expected_property_name);
     return 0;
   }
   return 1;
 }
 
+static int check_x509_pem_cert_chain(const grpc_auth_context* ctx,
+                                     const char* expected_pem_cert_chain) {
+  return check_property_with_single_value(
```

Comment:
Thanks for the refactoring! This looks much more concise.
For functions like this with only one return statement inside, can we just directly invoke `check_property_with_single_value(...)` at its original place instead of defining the same function again? 
Same for other `check_*` functions.